crates/pallet-executor/src/lib.rs

```diff
@@ -109,10 +117,37 @@ mod pallet {
 
             // TODO: ensure the receipt is ready to be applied
 
-            // TODO: apply the execution receipt.
+            let SignedExecutionReceipt {
+                execution_receipt, ..
+            } = signed_execution_receipt;
 
-            Self::deposit_event(Event::ExecutionReceiptStored {
-                receipt_hash: signed_execution_receipt.hash(),
+            let primary_hash = execution_receipt.primary_hash;
+            let primary_number = execution_receipt.primary_number;
+
+            // Execution receipt starts from the primary block #1.
+            if primary_number > 1u32 {
+                ensure!(
+                    Receipts::<T>::get(primary_number - 1u32).is_some(),
+                    Error::<T>::MissingParentReceipt
+                );
+            }
+
+            // Apply the execution receipt.
+            <Receipts<T>>::insert(primary_number, execution_receipt);
+            <ReceiptsRange<T>>::mutate(|range| {
+                range
+                    .get_or_insert_with(|| (primary_number, primary_number))
+                    .1 = primary_number + 1u32;
+            });
+
+            let pruning_depth = <ReceiptsPruningDepth<T>>::get();
+            if primary_number > pruning_depth {
```

Comment:
Something like this would describe clearer what you want to achieve:
```rust
if let Some(primary_number_to_prune) = primary_number.checked_sub(pruning_depth + 1) {
    Self::prune_receipts_up_to(primary_number_to_prune);
}

Reply:
Fine to me if that makes you feel better, updated in https://github.com/subspace/subspace/pull/427/commits/4f63a0332d137b43c4b2729f57c7b5344578719d